server/src/main/java/org/apache/iotdb/db/writelog/manager/ReadOnlyRecoverManager.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iotdb.db.writelog.manager;
+
+import org.apache.iotdb.db.common.RetryCounter;
+import org.apache.iotdb.db.common.RetryCounterFactory;
+import org.apache.iotdb.db.conf.IoTDBDescriptor;
+import org.apache.iotdb.db.engine.StorageEngine;
+import org.apache.iotdb.db.exception.RecoverReadOnlyException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/** read only mode recover default: 10 min once, retry 3 times */
+public class ReadOnlyRecoverManager {
+  private static final Logger logger = LoggerFactory.getLogger(ReadOnlyRecoverManager.class);
+
+  private static final int READ_ONLY_RECOVER_DEFAULT_RETRY_ATTEMPTS = 3;
+
+  private static final long READ_ONLY_RECOVER_DEFAULT_RETRY_SLEEP_IINTERVAL = 10 * 60 * 1000L;
```

Comment:
Hi, I think these two fields could be configured and users might like to change them.

Reply:
thanks @HTHou ,   I will add configuration in config...